pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the adapter class for Qualtran-PennyLane interoperability.
+"""
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import DecompositionUndefinedError, MatrixUndefinedError, Operation
+from pennylane.wires import WiresLike
+
+try:
+    import qualtran as qt
+except (ModuleNotFoundError, ImportError) as import_error:
+    pass
+
+
+def get_bloq_registers_info(bloq):
+    """Returns a ``qml.registers`` object associated with all named and unnamed registers and wires
+    in the bloq.
+
+    Args:
+        bloq (Bloq): the bloq to get the registers info of
+
+    Returns:
+        dict: A dictionary that has all the named and un-named registers with default wire
+        ordering.
+
+    Raises:
+        TypeError: bloq must be an instance of :code:`~.Bloq`.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    >>> from qualtran.bloqs.phase_estimation import RectangularWindowState, TextbookQPE
+    >>> textbook_qpe_small = TextbookQPE(ZPowGate(exponent=2 * 0.234), RectangularWindowState(3))
+    >>> qml.get_bloq_registers_info(textbook_qpe_small)
+    {'qpe_reg': Wires([0, 1, 2]), 'q': Wires([3])}
```

Comment:
What are the keys of this dictionary? :thinking: 
The Qualtran register names?

Reply:
Yes; if you defined your own composite bloq with your own names, they would show up as the keys to this dictionary. In this case, they're the default qualtran register names.